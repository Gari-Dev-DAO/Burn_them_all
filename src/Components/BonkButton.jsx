
const BonkButton = ({children,onClick,isDisabled}) => {
  return (
   <button onClick={onClick} disabled={isDisabled}  className='bonkBtn'>{children}</button>
  )
}

export default BonkButton
//render
//onclick should called
// if isDisabled than button should be disable else not