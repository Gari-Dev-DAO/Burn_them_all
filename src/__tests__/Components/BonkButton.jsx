import { fireEvent, render } from "@testing-library/react";
import BonkButton from "../../Components/BonkButton";

describe("BonkButton", () => {
  it("renders component", () => {
    const { container } = render(<BonkButton />);
    expect(container).toMatchSnapshot();
  });

  it("Onclick is working", () => {
    const onClick = jest.fn();
    const { getByText } = render(
      <BonkButton onClick={onClick}>Bonk</BonkButton>
    );
    const button = getByText("Bonk");
    fireEvent.click(button);
    expect(onClick).toBeCalledTimes(1);
  });

  it("Button disables based on props", () => {
    const isDisabled = true;
    const { getByText } = render(
      <BonkButton isDisabled={isDisabled}>Bonk</BonkButton>
    );
    const button = getByText("Bonk");
    expect(button).toBeDisabled();
  });

  it("Button enables based on props", () => {
    const isDisabled = false;
    const { getByText } = render(
      <BonkButton isDisabled={isDisabled}>Bonk</BonkButton>
    );
    const button = getByText("Bonk");
    expect(button).not.toBeDisabled();
  });
});
