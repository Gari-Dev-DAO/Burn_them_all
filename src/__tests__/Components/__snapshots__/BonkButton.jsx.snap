// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BonkButton renders component 1`] = `
<div>
  <button
    class="bonkBtn"
  />
</div>
`;
